native/src/routes/offers/containers/OffersContainer.js

```diff
@@ -94,17 +84,20 @@ const OffersContainer = ({ theme, t, navigation, route }: OffersPropsType) => {
     [offers, languageCode, cityCode, navigation]
   )
 
-  if (error) {
+  const cityModel = cities && cities.find(city => city.code === cityCode)
+  if (offersError || (cityModel && !cityModel.offersEnabled)) {
+    const error =
+      offersError || new NotFoundError({ type: 'category', id: 'offers', city: cityCode, language: languageCode })
     return (
-      <LayoutedScrollView refreshControl={<RefreshControl onRefresh={loadOffers} refreshing={loading} />}>
-        <FailureContainer code={fromError(error)} tryAgain={tryAgain} />
+      <LayoutedScrollView refreshControl={<RefreshControl onRefresh={refresh} refreshing={loading} />}>
+        <FailureContainer code={fromError(error)} tryAgain={refresh} />
       </LayoutedScrollView>
     )
   }
 
   return (
-    <LayoutedScrollView refreshControl={<RefreshControl onRefresh={loadOffers} refreshing={loading} />}>
-      {offers && (
+    <LayoutedScrollView refreshControl={<RefreshControl onRefresh={refresh} refreshing={loading} />}>
+      {cities && offers && (
```

Comment:
So shouldn't there be a `NotFoundError` if cities are not defined?

Reply:
Hmmm normally it isn't possible that cities are undefined here as the fetch failed earlier already. I'd say if there are no cities it is not a NotFoundError.

Reply:
Hmm yeah good question. If there are no cities it is not possible to check whether the feature is enabled. But I guess I'll just remove it as this edge case should not happen at all anyway.